interfaces/builtin/adb_support.go

```diff
@@ -0,0 +1,205 @@
+	// MOTOROLA
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="22b8", MODE="0666"`,
+	// MSI
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="0db0", MODE="0666"`,
+	// MTK
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="0e8d", MODE="0666"`,
+	// NEC
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="0409", MODE="0666"`,
+	// NOOK
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="2080", MODE="0666"`,
+	// NVIDIA
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="0955", MODE="0666"`,
+	// OPPO
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="22d9", MODE="0666"`,
+	// OTGV
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="2257", MODE="0666"`,
+	// OUYA
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="2836", MODE="0666"`,
+	// PANTECH
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="10a9", MODE="0666"`,
+	// PEGATRON
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="1d4d", MODE="0666"`,
+	// PHILPS
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="0471", MODE="0666"`,
+	// PMC-SIERRA
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="04da", MODE="0666"`,
+	// POSITIVO
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="1662", MODE="0666"`,
+	// PRESTIGIO
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="29e4", MODE="0666"`,
+	// QISDA
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="1d45", MODE="0666"`,
+	// Qualcomm
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="05c6", MODE="0666"`,
+	// QUANTA
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="0408", MODE="0666"`,
+	// ROCKCHIP
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="2207", MODE="0666"`,
+	// SAMSUNG
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="04e8", MODE="0666"`,
+	// SHARP
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="04dd", MODE="0666"`,
+	// SK TELESYS
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="1f53", MODE="0666"`,
+	// SMARTISAN
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="29a9", MODE="0666"`,
+	// SONY
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="054c", MODE="0666"`,
+	// SONY ERICSSON
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="0fce", MODE="0666"`,
+	// T_AND_A
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="1bbb", MODE="0666"`,
+	// TECHFAITH
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="1d09", MODE="0666"`,
+	// TELEEPOCH
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="2340", MODE="0666"`,
+	// TI
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="0451", MODE="0666"`,
+	// TOSHIBA
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="0930", MODE="0666"`,
+	// UNOWHY
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="2a49", MODE="0666"`,
+	// VIZIO
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="E040", MODE="0666"`,
+	// WACOM
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="0531", MODE="0666"`,
+	// XIAOMI
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="2717", MODE="0666"`,
+	// YOTADEVICES
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="2916", MODE="0666"`,
+	// YULONG_COOLPAD
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="1ebf", MODE="0666"`,
+	// ZTE
+	`SUBSYSTEM=="usb", ATTR{idVendor}=="19d2", MODE="0666"`,
+}
+
+func init() {
+	registerIface(&commonInterface{
+		name:                 "adb-support",
+		summary:              adbSupportSummary,
+		implicitOnCore:       true,
+		implicitOnClassic:    true,
+		baseDeclarationSlots: adbSupportBaseDeclarationSlots,
+		connectedPlugUDev:    adbSupportConnectedPlugUDev,
```

Comment:
If we go with the slot representation, this should be:
```
implicitOnCore: false,
implicitOnClassic: true,
permanentSlotUDev: adbSupportPermanentSlotUDev,
```
(you'll of course need to adjust the variable name up above).

What we'll want to verify that the registerInterface actually achieves that and that we don't need to override UDevPermanentSlot with a `if !release.OnClassic { ... <add udev rules> ... }`